src/client/app/graph/card/card.component.ts

```diff
@@ -30,12 +38,17 @@ export class CardComponent implements OnInit {
   // The last point seen during the drag that is currently in progress.
   private lastDragPnt: Point = null; //TODO(eyuelt): make this nullable after TS2 update
 
-  constructor(private canvasService: CanvasService) {
+  constructor(private canvasService: CanvasService,
+              private googleRealtimeService: GoogleRealtimeService) {
   }
 
   ngOnInit() {
     this.position = this.canvasService.canvasCoordToViewportCoord(this.card);
     this.canvasService.addListener(this.update.bind(this));
+    this.googleRealtimeService.currentDocument.subscribe(document => {
+      document.getModel().getRoot()
+          .addEventListener(OBJECT_CHANGED, this.update.bind(this));
```

Comment:
Add a TODO to change this to only listen to events on this card, rather than on the whole document.